test/OptionalObjectiveTest.js

```diff
@@ -0,0 +1,165 @@
+/*global require*/
+/*global describe*/
+/*global it*/
+
+var assert = require("assert");
+var solver = require("../src/main.js");
+
+describe("Testing if optional objectives are taken into account", function () {
+
+	it("should be able to construct and solve UI model", function () {
+
```

Comment:
May be a short comment here looking like this: "The following model describes a small UI (User Interface) problem. Variables x, y and w, h respectively correspond to positions and dimensions of a UI element."